### PR TITLE
Decouple accounts hash calculation from snapshot hash

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Accounts hash and snapshot hash are tied to the same rate, but the node operator may want to create snapshots at a different rate. Also accounts cleaning functions are tied to creating snapshots and not being run otherwise.

#### Summary of Changes

Decouple the snapshot creation rate and accounts clean and accounts hashing rate.

Fixes #

### CODE REVIEW

**Comment by mvines** [comment]
@sakridge - are you thinking of making the accounts hashing rate slower than snapshot hash rate?  I ask because I started depending on accounts hashing for the RPC health check (#9499)

**Comment by sakridge** [comment]
> 
> 
> @sakridge - are you thinking of making the accounts hashing rate slower than snapshot hash rate? I ask because I started depending on accounts hashing for the RPC health check (#9499)

Ideally faster than snapshot rate.

**Comment by mvines** [comment]
> Ideally faster than snapshot rate.
Perfect!

File: ledger/src/bank_forks.rs [hunk 1]
```diff
@@ -209,43 +216,46 @@ impl BankForks {
...
+        for bank in banks.iter() {
+            let bank_slot = bank.slot();
+            if bank.block_height() % self.accounts_hash_interval_slots == 0
+                && bank_slot > self.last_accounts_hash_slot
+            {
+                self.last_accounts_hash_slot = bank_slot;
+                bank.squash();
+                is_root_bank_squashed = bank_slot == root;
+
+                bank.clean_accounts();
```
Comment (ryoqun) [inline ledger/src/bank_forks.rs:233]: These calls are needed for `ledger-tool create-snapshot` as well.... #10212

**Pull request merged**